internal/proto/data_coord.proto

```diff
@@ -828,3 +842,52 @@ message GcControlRequest {
   GcCommand command = 2;
   repeated common.KeyValuePair params = 3;
 }
+
+enum CompactionTaskState {
+  unknown = 0;
+  compact_executing = 1; // => executing
+  compact_pipelining = 2;
+  compact_completed = 3; // => completed
+  failed = 4;
+  timeout = 5;
+  init = 6;
+  analyzing = 7;
+  analyzed = 8;
+  indexing = 9;
+  indexed = 10;
+  cleaned = 11;
+}
+
+message CompactionTask {
```

Comment:
maybe `ClusteringCompactionTask` is better

Reply:
This is for future. If we store a meta for all types of compaction.